src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,614 @@
+              .ChildNodes.Should().ContainSingle<HttpHeadersNode>().Which;
+
+        var headerNodes = headersNode.HeaderNodes.ToArray();
+        headerNodes.Should().HaveCount(5);
+
+        headerNodes[0].NameNode.Text.Should().Be("Accept");
+        headerNodes[0].SeparatorNode.Text.Should().Be(":");
+        headerNodes[0].ValueNode.Text.Should().Be("*/*");
+
+        headerNodes[1].NameNode.Text.Should().Be("Accept-Encoding");
+        headerNodes[1].SeparatorNode.Text.Should().Be(":");
+        headerNodes[1].ValueNode.Text.Should().Be("gzip, deflate, br");
+
+        headerNodes[2].NameNode.Text.Should().Be("Accept-Language");
+        headerNodes[2].SeparatorNode.Text.Should().Be(":");
+        headerNodes[2].ValueNode.Text.Should().Be("en-US,en;q=0.9");
+
+        headerNodes[3].NameNode.Text.Should().Be("ContentLength");
+        headerNodes[3].SeparatorNode.Text.Should().Be(":");
+        headerNodes[3].ValueNode.Text.Should().Be("7060");
+
+        headerNodes[4].NameNode.Text.Should().Be("Cookie");
+        headerNodes[4].SeparatorNode.Text.Should().Be(":");
+        headerNodes[4].ValueNode.Text.Should().Be("expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;");
+    }
+
+    [Fact]
+    public void header_with_body_is_parsed_correctly()
+    {
+        var result = HttpRequestParser.Parse(
+            """
+            POST https://example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            Authorization: token xxx
+
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+
+        using var _ = new AssertionScope();
+
+        var requestNode = result.SyntaxTree.RootNode
+              .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+        requestNode.HeadersNode.HeaderNodes.Count.Should().Be(2);
+        requestNode.BodySeparatorNode.TextWithTrivia.Should().Be("\r\n");
+        requestNode.BodyNode.Text.Should().Be(
+            """
+            <request>
+                <name>sample</name>
+                <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+            </request>
+            """);
+    }
+
+    [Fact]
+    public void body_without_header_header_is_parsed_correctly()
+    {
+        var result = HttpRequestParser.Parse(
+            """
+        POST https://example.com/comments HTTP/1.1
+
+        <request>
+            <name>sample</name>
+            <time>Wed, 21 Oct 2015 18:27:50 GMT</time>
+        </request>
+        """);
+
+        using var _ = new AssertionScope();
+
+        var requestNode = result.SyntaxTree.RootNode
+              .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which;
+
+        requestNode.HeadersNode.HeaderNodes.Count.Should().Be(0);
+    }
+
+    //TODO: Is it an error to include a body with no headers?
+    [Fact]
+    public void body_without_header_body_is_parsed_correctly()
```

Comment:
The names of this test and the one above are confusing. Could you rename them to match what is being asserted?